website/docs/r/kendra_index.html.markdown

```diff
@@ -0,0 +1,172 @@
+---
+subcategory: "Kendra"
+layout: "aws"
+page_title: "AWS: aws_kendra_index"
+description: |-
+  Provides details about a specific Amazon Kendra Index.
+---
+
+# Resource: aws_kendra_index
+
+## Example Usage
+
+### Basic
+
+```terraform
+resource "aws_kendra_index" "example" {
+  name        = "example"
+  description = "example"
+  edition     = "DEVELOPER_EDITION"
+  role_arn    = aws_iam_role.this.arn
+
+  tags = {
+    "Key1" = "Value1"
+  }
+}
+```
+
+### With server side encryption configuration
+
+```terraform
+resource "aws_kendra_index" "example" {
+  name     = "example"
+  role_arn = aws_iam_role.this.arn
+
+  server_side_encryption_configuration {
+    kms_key_id = data.aws_kms_key.this.arn
+  }
+}
+```
+
+### With JSON token type configuration
+
+```terraform
+resource "aws_kendra_index" "example" {
+  name     = "example"
+  role_arn = aws_iam_role.this.arn
+
+  user_token_configurations {
+    json_token_type_configuration {
+      group_attribute_field     = "groups"
+      user_name_attribute_field = "username"
+    }
+  }
+}
+```
+
+## Argument Reference
+
+### Arguments supported at create time
```

Comment:
Nit: we might not need this extra header if it looks like all the configurable params are supported at create time

Reply:
Thanks @anGie44! I was wondering about this because there are 2 params that are only supported by the `UpdateIndex` call. `UpdateIndex` supports arguments like [CapacityUnits](https://docs.aws.amazon.com/kendra/latest/dg/API_UpdateIndex.html#API_UpdateIndex_RequestSyntax) and [DocumentMetadataConfigurationUpdates](https://docs.aws.amazon.com/kendra/latest/dg/API_UpdateIndex.html#API_UpdateIndex_RequestSyntax) that CreateIndex does not support. I was thinking these 2 can go into a separate resource?

Reply:
Ohh i see. Interesting, hmm yeah that gets a bit tricky perhaps because going by the acceptance tests, the API sends back some default values for these 2 params. I'll have to look at some of our other resources and see what we do in cases like this

Reply:
but i do like the idea of having separate resources!